[#80] Merge the hydroshare 1.14.1 branch.

dsummersl opened the pull request:
I'd suggest the following method of testing this:

```bash
# stop and remove whatever docker setup you currently have:
docker-compose stop
docker-compose rm

# rebuild the docker images and the docker database from the repository:
./hsctl rebuild --db
```
Then go to http://localhost:8000 and try out all the common areas of the site.

dsummersl committed:
[#80] Merge the hydroshare 1.14.1 branch.

cbcunc commented:
Alternatively, you can
`./hsctl stop`
and either of
`scripts\cleandocker`
or
`./hsctl reset_all`

philtuskha commented on config/hydroshare-config.yaml:
```diff
@@ -1,8 +1,8 @@
-HOST_SSL_DIR: /home/hydro/hs-certs
+HS_PATH: /home/dsummers/rs/CK/hydroshare/MyHPOM
+HOST_SSL_DIR: /home/dsummers/rs/CK/hydroshare/hs-certs
 HS_DATABASE: pg.myhpomdevelopment.sql
```
Looks like some of your personal settings for your purely Linux environment got encluded.

dsummersl replied:
ooh, good catch, thanks...

philtuskha reviewed (changes requested):
Besides those local setup changes, this all looks good to my bleary eyes and seems to work fine.

dsummersl committed:
[#80] Removes local dev config settings.

philtuskha reviewed (approved):
👍

cbcunc commented:
Revert changes to hydroshare/local_settings.py

dsummersl committed:
Merge branch 'xdci-develop' into 80-merge-hydroshare-1.14-release

dsummersl committed:
[#80] Remove organization from the sign up form.

dsummersl committed:
[#80] Removes select document from upload content page.

dsummersl committed:
[#80] Removes commented out code.

dsummersl committed:
[#80] Removes hydroshare specific link on uploading.

dsummersl committed:
[#80] Adds imports that could be needed.

But probably won't with the current way organization is disabled.

dsummersl committed:
[#80] Restores nag message.

I don't think we want this b/c it appears to only nag you if your
organization has not been supplied.

dsummersl committed:
[#80] Removes Basic authentication from DRF.

Since session based authentication is still in place, and the site isn't
meant to support a REST framework except for the site itself (which
will use session information), I think basic authentication isn't
needed.

This also allows basic authentication site wide for staging (before
django rest framework would erroneously throw an error b/c the Basic
Auth configured for nginx would not match any known user on the site).

Possibly another workaround for this would be to have nginx strip the
http basic authentication from any requests that were sent along to the
django server.

dsummersl committed:
[#80] Temporarily disable organization nag message.

We hope to use this to better affect once we've added more fields to the
profile page.

dsummersl committed:
[#80] Fixes test groups to 'resources'.

The test cases were using 'hydroshare' groups. I suspect that there are
test cases we might have modified long ago that broke these tests that
we could/should revert at some point. In the meantime this is a quick
fix.

dsummersl committed:
[#80] Fikes myhpom quota tests.

Our zones are myhpom related, not hydroshare related. This fixes a
number of tests.

dsummersl committed:
[#80] Fixes badly merged test.

The last merge looks to have missed some formatting/try-catch blocks.

dsummersl committed:
[#80] Fixes error message assertions.

I'm not clear on exactly why these are different in myhpom, but they are
slightly different.

dsummersl committed:
[#80] Removes unnecessary myhpom comments.

dsummersl committed:
[#80] Fixes API to use generic resources.

Netcdf aren't used in myhpom.

dsummersl committed:
[#80] Disables tests for unused feature in myhpom.

dsummersl committed:
Merge branch 'xdci-develop' into 80-merge-hydroshare-1.14-release
